src/Executables/Examples/SingletonTovProfile/SingletonTovProfile.hpp

```diff
@@ -0,0 +1,207 @@
+        Parallel::get<TovSingletonCacheTags::CentralMassDensity>(cache);
+    const size_t num_pts =
+        Parallel::get<TovSingletonCacheTags::NumberOfPoints>(cache);
+    const auto& equation_of_state =
+        Parallel::get<OptionTags::EquationOfState<true, 1>>(cache);
+
+    // Integrate TOV equations out to the surface of the star:
+    const double surface_log_enthalpy = 0.0;
+    const gr::Solutions::TovSolution tov_star(
+        equation_of_state, central_mass_density, surface_log_enthalpy);
+    const double outer_radius = tov_star.outer_radius();
+
+    // Define helper closures:
+    const auto pressure_from_density = [&equation_of_state](
+        const double density) noexcept {
+      return equation_of_state.pressure_from_density(Scalar<double>{density})
+          .get();
+    };
+    const auto energy_from_density = [&equation_of_state](
+        const double density) noexcept {
+      return equation_of_state
+          .specific_internal_energy_from_density(Scalar<double>{density})
+          .get();
+    };
+    const auto sound_speed_squared_from_density = [&equation_of_state](
+        const double density) noexcept {
+      return equation_of_state.chi_from_density(Scalar<double>{density}).get() +
+             equation_of_state
+                 .kappa_times_p_over_rho_squared_from_density(
+                     Scalar<double>{density})
+                 .get();
+    };
+
+    // Detailed output is written to the H5 file:
+    const std::string h5_file_name = filename + ".h5";
+    if (file_system::check_if_file_exists(h5_file_name)) {
+      file_system::rm(h5_file_name, true);
+    }
+    h5::H5File<h5::AccessType::ReadWrite> my_file(h5_file_name);
+    std::vector<std::string> legend{
+        "Radius", "Mass",         "Pressure",           "Density",
+        "Energy", "Log Enthalpy", "Sound speed squared"};
+    const uint32_t version_number = 0;
+    auto& tov_dat_file =
+        my_file.insert<h5::Dat>("/TovProfile", legend, version_number);
+    const double delta_r_finer = outer_radius / (num_pts - 1.0);
+    for (size_t i = 0; i < num_pts; i++) {
+      const auto current_radius = i * delta_r_finer;
+      const auto enclosed_mass = tov_star.mass(current_radius);
+      const auto log_enthalpy = tov_star.log_specific_enthalpy(current_radius);
+      const auto enthalpy = std::exp(log_enthalpy);
+      const auto density = (equation_of_state.rest_mass_density_from_enthalpy(
+                                Scalar<double>{enthalpy}))
+                               .get();
+      const auto pressure = pressure_from_density(density);
+      const auto energy = energy_from_density(density);
+      const auto sound_speed_squared =
+          sound_speed_squared_from_density(density);
+      std::vector<double> row_of_tov_data{
+          current_radius, enclosed_mass, pressure,           density,
+          energy,         log_enthalpy,  sound_speed_squared};
+      tov_dat_file.append(row_of_tov_data);
```

Comment:
It should be better to create a `std::vector<std::vector<double>>` and then append that once to the file so you do less IO.